sdk/dapp-kit/src/hooks/wallet/useCurrentWallet.ts

```diff
@@ -7,5 +7,46 @@ import { useWalletStore } from './useWalletStore.js';
  * Retrieves the wallet that is currently connected to the dApp, if one exists.
  */
 export function useCurrentWallet() {
-	return useWalletStore((state) => state.currentWallet);
+	const currentWallet = useWalletStore((state) => state.currentWallet);
+	const connectionStatus = useWalletStore((state) => state.connectionStatus);
+
+	switch (connectionStatus) {
```

Comment:
Not a blocking concern by any means, just a question, but why not just individually compute these properties, then put them on an object in a single return vs doing this switch?

Reply:
I did the branched `as const` returns to get `react-query` style type narrowing, if there's an easier way to do this then please do tell hahaha

```
if (isConnected) {
  console.log(currentWallet, '<-- this is not null');
}
```

Reply:
Oh that makes sense! That's totally fine I was just curious! :)